ERROR: Failed building wheel for llama-cpp-python

Description:
I encountered an error while building the llama_cpp_python package during a Podman build to create the container image. The error occurs during the wheel building process and fails. Here is a part of the output:

First part:

      `Building wheels for collected packages: llama-cpp-python
        Building wheel for llama-cpp-python (pyproject.toml): started
        Building wheel for llama-cpp-python (pyproject.toml): finished with status 'error'
        error: subprocess-exited-with-error
      
        × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
        │ exit code: 1
        ╰─> [286 lines of output]
            *** scikit-build-core 0.9.6 using CMake 3.29.6 (wheel)
            *** Configuring CMake...
            loading initial cache file /tmp/tmpvfm1_3w0/build/CMakeInit.txt
            -- The C compiler identification is GNU 11.4.1
            -- The CXX compiler identification is GNU 11.4.1
            -- Detecting C compiler ABI info
            -- Detecting C compiler ABI info - done
            -- Check for working C compiler: /usr/bin/cc - skipped
            -- Detecting C compile features
            -- Detecting C compile features - done
            -- Detecting CXX compiler ABI info
            -- Detecting CXX compiler ABI info - done
            -- Check for working CXX compiler: /usr/bin/c++ - skipped
            -- Detecting CXX compile features
            -- Detecting CXX compile features - done
            -- Found Git: /usr/bin/git (found version "2.43.5")
            -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
            -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
            -- Found Threads: TRUE
            CMake Warning at vendor/llama.cpp/CMakeLists.txt:390 (message):
              LLAMA_CUBLAS is deprecated and will be removed in the future.
      
              Use LLAMA_CUDA instead`


Second part:



    ` [49/56] /usr/local/cuda/bin/nvcc -forward-unknown-to-host-compiler -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_MMV_Y=1 -DGGML_CUDA_PEER_MAX_BATCH_SIZE=128 -DGGML_CUDA_USE_GRAPHS -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_CUDA -DGGML_USE_LLAMAFILE -DK_QUANTS_PER_ITERATION=2 -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-wnwch3cz/llama-cpp-python_ca537cafc006490daa830165e9666ecc/vendor/llama.cpp/. -isystem /usr/local/cuda/targets/x86_64-linux/include -O3 -DNDEBUG -std=c++11 "--generate-code=arch=compute_52,code=[compute_52,sm_52]" "--generate-code=arch=compute_61,code=[compute_61,sm_61]" "--generate-code=arch=compute_70,code=[compute_70,sm_70]" -Xcompiler=-fPIC -use_fast_math -Xcompiler "-Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wno-format-truncation -Wextra-semi -Wno-pedantic -march=native" -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-cuda/mmvq.cu.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-cuda/mmvq.cu.o.d -x cu -c /tmp/pip-install-wnwch3cz/llama-cpp-python_ca537cafc006490daa830165e9666ecc/vendor/llama.cpp/ggml-cuda/mmvq.cu -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-cuda/mmvq.cu.o
          ninja: build stopped: subcommand failed.
    
    
          *** CMake build failed
          [end of output]
    
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for llama-cpp-python
    ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (llama-cpp-python)
    Error: building at STEP "RUN CMAKE_ARGS="-DLLAMA_CUBLAS=on" FORCE_CMAKE=1 python3.11 -m pip install -c /tmp/constraints.txt --force-reinstall --no-cache-dir `

Steps to Reproduce:

1. Use Podman to build the container image which includes the llama_cpp_python package.
2. During the build process, the error occurs while building the wheel for llama_cpp_python.

Expected Behavior:
The wheel for llama_cpp_python should be built successfully without any errors.

Actual Behavior:
The build process fails with a subprocess-exited-with-error status during the wheel building for llama_cpp_python.

Additional Information:
 - Podman version: 4.9.3
 - OS Version: Fedora 39
 - Python Version: Like in the containerfile 3.11

Please let me know if you need any further information to help diagnose this issue.

Thank you!

I may have been able to reproduce it, here is the output from `tox -e mypy`:

```
Building wheels for collected packages: llama-cpp-python
  Building wheel for llama-cpp-python (pyproject.toml) ... error
Failed to build llama-cpp-python
  error: subprocess-exited-with-error

  × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [92 lines of output]
      *** scikit-build-core 0.9.6 using CMake 3.29.6 (wheel)
      *** Configuring CMake...
      loading initial cache file /tmp/tmpvns_a0b4/build/CMakeInit.txt
      -- The C compiler identification is GNU 14.1.1
      -- The CXX compiler identification is GNU 14.1.1
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Git: /usr/bin/git (found version "2.45.2")
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
      -- Found Threads: TRUE
      -- Warning: ccache not found - consider installing it for faster compilation or disable this warning with LLAMA_CCACHE=OFF
      -- CMAKE_SYSTEM_PROCESSOR: x86_64
      -- x86 detected
      CMake Warning (dev) at CMakeLists.txt:26 (install):
        Target llama has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      This warning is for project developers.  Use -Wno-dev to suppress it.

      CMake Warning (dev) at CMakeLists.txt:35 (install):
        Target llama has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      This warning is for project developers.  Use -Wno-dev to suppress it.

      -- Configuring done (0.8s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/tmpvns_a0b4/build
      *** Building project with Ninja...
      Change Dir: '/tmp/tmpvns_a0b4/build'

      Run Build Command(s): /tmp/pip-build-env-kdhkkv09/normal/lib64/python3.12/site-packages/ninja/data/bin/ninja -v
      [1/27] /usr/bin/cc -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu11 -fPIC -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wdouble-promotion -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-alloc.c.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-alloc.c.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-alloc.c.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-alloc.c
      [2/27] /usr/bin/cc -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu11 -fPIC -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wdouble-promotion -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-backend.c.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-backend.c.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-backend.c.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-backend.c
      [3/27] /usr/bin/cc -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu11 -fPIC -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wdouble-promotion -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c
      FAILED: vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o
      /usr/bin/cc -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu11 -fPIC -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wdouble-promotion -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml-quants.c.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c
      In file included from /usr/lib/gcc/x86_64-redhat-linux/14/include/immintrin.h:117,
                       from /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-impl.h:435,
                       from /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c:5:
      /usr/lib/gcc/x86_64-redhat-linux/14/include/fmaintrin.h: In function ‘ggml_vec_dot_q4_0_q8_0’:
      /usr/lib/gcc/x86_64-redhat-linux/14/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
         63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
            | ^~~~~~~~~~~~~~~
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c:3621:15: note: called from here
       3621 |         acc = _mm256_fmadd_ps( d, q, acc );
            |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /usr/lib/gcc/x86_64-redhat-linux/14/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
         63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
            | ^~~~~~~~~~~~~~~
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c:3621:15: note: called from here
       3621 |         acc = _mm256_fmadd_ps( d, q, acc );
            |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /usr/lib/gcc/x86_64-redhat-linux/14/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
         63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
            | ^~~~~~~~~~~~~~~
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c:3621:15: note: called from here
       3621 |         acc = _mm256_fmadd_ps( d, q, acc );
            |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /usr/lib/gcc/x86_64-redhat-linux/14/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
         63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
            | ^~~~~~~~~~~~~~~
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml-quants.c:3621:15: note: called from here
       3621 |         acc = _mm256_fmadd_ps( d, q, acc );
            |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      [4/27] /usr/bin/c++ -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu++11 -fPIC -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/sgemm.cpp.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/sgemm.cpp.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/sgemm.cpp.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/sgemm.cpp
      [5/27] /usr/bin/c++ -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -DLLAMA_BUILD -DLLAMA_SHARED -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -Dllama_EXPORTS -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu++11 -fPIC -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native -MD -MT vendor/llama.cpp/CMakeFiles/llama.dir/unicode-data.cpp.o -MF vendor/llama.cpp/CMakeFiles/llama.dir/unicode-data.cpp.o.d -o vendor/llama.cpp/CMakeFiles/llama.dir/unicode-data.cpp.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/unicode-data.cpp
      [6/27] /usr/bin/c++ -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -DLLAMA_BUILD -DLLAMA_SHARED -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -Dllama_EXPORTS -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu++11 -fPIC -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native -MD -MT vendor/llama.cpp/CMakeFiles/llama.dir/unicode.cpp.o -MF vendor/llama.cpp/CMakeFiles/llama.dir/unicode.cpp.o.d -o vendor/llama.cpp/CMakeFiles/llama.dir/unicode.cpp.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/unicode.cpp
      [7/27] /usr/bin/cc -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu11 -fPIC -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wdouble-promotion -march=native -MD -MT vendor/llama.cpp/CMakeFiles/ggml.dir/ggml.c.o -MF vendor/llama.cpp/CMakeFiles/ggml.dir/ggml.c.o.d -o vendor/llama.cpp/CMakeFiles/ggml.dir/ggml.c.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c: In function ‘ggml_vec_mad_f16’:
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c:1868:45: warning: passing argument 1 of ‘__avx_f32cx8_load’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
       1868 |             ax[j] = GGML_F16_VEC_LOAD(x + i + j*GGML_F16_EPR, j);
            |                                             ^
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c:1238:51: note: in definition of macro ‘GGML_F32Cx8_LOAD’
       1238 | #define GGML_F32Cx8_LOAD(x)     __avx_f32cx8_load(x)
            |                                                   ^
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c:1868:21: note: in expansion of macro ‘GGML_F16_VEC_LOAD’
       1868 |             ax[j] = GGML_F16_VEC_LOAD(x + i + j*GGML_F16_EPR, j);
            |                     ^~~~~~~~~~~~~~~~~
      /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/ggml.c:1221:53: note: expected ‘ggml_fp16_t *’ {aka ‘short unsigned int *’} but argument is of type ‘const ggml_fp16_t *’ {aka ‘const short unsigned int *’}
       1221 | static inline __m256 __avx_f32cx8_load(ggml_fp16_t *x) {
            |                                        ~~~~~~~~~~~~~^
      [8/27] /usr/bin/c++ -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_LLAMAFILE -DLLAMA_BUILD -DLLAMA_SHARED -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -Dllama_EXPORTS -I/tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/. -O3 -DNDEBUG -std=gnu++11 -fPIC -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native -MD -MT vendor/llama.cpp/CMakeFiles/llama.dir/llama.cpp.o -MF vendor/llama.cpp/CMakeFiles/llama.dir/llama.cpp.o.d -o vendor/llama.cpp/CMakeFiles/llama.dir/llama.cpp.o -c /tmp/pip-install-l1677iax/llama-cpp-python_507ca0338be7449ab98de4b2d78ee978/vendor/llama.cpp/llama.cpp
      ninja: build stopped: subcommand failed.


      *** CMake build failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

mypy: exit 1 (100.12 seconds) /home/vagrant/git/instructlab> python -I -m pip install 'GitPython>=3.1.42' 'PyYAML>=6.0.0' 'click-didyoumean>=0.3.0' 'click<9.0.0,>=8.1.7' 'datasets>=2.18.0' 'gguf>=0.6.0' 'httpx>=0.25.0' 'instructlab-quantize>=0.1.0' 'instructlab-schema>=0.2.0' 'instructlab-sdg>=0.0.2' 'instructlab-training>=0.0.3' 'jsonschema>=4.21.1' langchain-text-splitters 'llama-cpp-python[server]==0.2.75' 'mlx<0.6.0,>=0.5.1; sys_platform == "darwin" and platform_machine == "arm64"' 'numpy<2.0.0,>=1.23.5; python_version == "3.10"' 'numpy<2.0.0,>=1.26.4; python_version != "3.10"' 'openai>=1.13.3' 'peft>=0.9.0' 'prompt-toolkit>=3.0.38' 'pydantic-yaml>=1.2.0' 'pydantic>=2.7.4' 'referencing<0.36.0,>=0.35.0' 'rich>=13.3.1' 'rouge-score>=0.1.2' 'sentencepiece>=0.2.0' 'tenacity!=8.4.0,>=8.3.0' 'tokenizers>=0.11.1' 'toml>=0.10.2' 'torch<3.0.0,>=2.3.0; python_version != "3.10"' 'torch<3.0.0,>=2.3.0a0; python_version == "3.10"' 'tqdm>=4.66.2' 'transformers>=4.41.2' 'trl>=0.9.4' 'wandb>=0.16.4' 'yamllint>=1.35.1' pid=3139
  mypy: FAIL code 1 (103.93 seconds)
  evaluation failed :( (104.03 seconds)
```

--- a/README.md
+++ b/README.md
@@ -177,7 +177,9 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
       python3 -m venv --upgrade-deps venv
       source venv/bin/activate
       pip cache remove llama_cpp_python
-      pip install instructlab -C cmake.args="-DLLAMA_CUDA=on" -C cmake.args="-DLLAMA_NATIVE=off"
+      pip install instructlab \
+         -C cmake.args="-DLLAMA_CUDA=on" \
+         -C cmake.args="-DLLAMA_NATIVE=off"
    ```
 
 4. From your `venv` environment, verify `ilab` is installed correctly, by running the `ilab` command.

--- a/README.md
+++ b/README.md
@@ -90,7 +90,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
 - Python 3.10 or Python 3.11
 - Approximately 60GB disk space (entire process)
 
-> **NOTE:** PyTorch 2.2.1 does not support `torch.compile` with Python 3.12. On Fedora 39+, install `python3.11-devel` and create the virtual env with `python3.11` if you wish to use PyTorch's JIT compiler.
+> **NOTE:** Python 3.12 is currently not supported, because some dependencies don't work on Python 3.12, yet.
 <!-- -->
 > **NOTE:** When installing the `ilab` CLI on macOS, you may have to run the `xcode select --install` command, installing the required packages previously listed.
 
@@ -98,16 +98,10 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
 
 ### 🧰 Installing `ilab`
 
-1. When installing on Fedora Linux, install C++, Python 3.10+, and other necessary tools by running the following command:
+1. When installing on Fedora Linux, install C++, Python 3.10 or 3.11, and other necessary tools by running the following command:
 
    ```shell
-   sudo dnf install g++ gcc make pip python3 python3-devel python3-GitPython
-   ```
-
-   Optional: If `g++` is not found, try `gcc-c++` by running the following command:
-
-   ```shell
-   sudo dnf install gcc-c++ gcc make pip python3 python3-devel python3-GitPython
+   sudo dnf install gcc gcc-c++ make git python3.11 python3.11-devel
    ```
 
    If you are running on macOS, this installation is not necessary and you can begin your process with the following step.
@@ -135,7 +129,9 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
       python3 -m venv --upgrade-deps venv
       source venv/bin/activate
       pip cache remove llama_cpp_python
-      pip install instructlab --extra-index-url=https://download.pytorch.org/whl/cpu
+      pip install instructlab \
+         --extra-index-url=https://download.pytorch.org/whl/cpu \
+         -C cmake.args="-DLLAMA_NATIVE=off"
       ```
 
       > **NOTE**: *Additional Build Argument for Intel Macs*
@@ -158,7 +154,8 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
          -C cmake.args="-DAMDGPU_TARGETS=all" \
          -C cmake.args="-DCMAKE_C_COMPILER=/opt/rocm/llvm/bin/clang" \
          -C cmake.args="-DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++" \
-         -C cmake.args="-DCMAKE_PREFIX_PATH=/opt/rocm"
+         -C cmake.args="-DCMAKE_PREFIX_PATH=/opt/rocm" \
+         -C cmake.args="-DLLAMA_NATIVE=off"
       ```
 
       On Fedora 40+, use `-DCMAKE_C_COMPILER=clang-17` and `-DCMAKE_CXX_COMPILER=clang++-17`.
@@ -180,7 +177,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
       python3 -m venv --upgrade-deps venv
       source venv/bin/activate
       pip cache remove llama_cpp_python
-      pip install instructlab -C cmake.args="-DLLAMA_CUDA=on"
+      pip install instructlab -C cmake.args="-DLLAMA_CUDA=on" -C cmake.args="-DLLAMA_NATIVE=off"
    ```
 
 4. From your `venv` environment, verify `ilab` is installed correctly, by running the `ilab` command.

